dashboard/app/controllers/projects_controller.rb

```diff
@@ -309,6 +309,9 @@ def show
     no_footer = sharing
     # if the game doesn't own the sharing footer, treat it as a legacy share
     @legacy_share_style = sharing && !@game.owns_footer_for_share?
+    if params[:key] == "spritelab"
+      @game.name = "Spritelab"
```

Comment:
Rather than overriding the game name (but still referencing the original `Game` instance) I wonder if we should be adding a proper Spritelab game type?

https://github.com/code-dot-org/code-dot-org/blob/e1ba61fa5fa32c02d7e74bc493519fbd6504ea2f/dashboard/app/models/levels/gamelab_jr.rb#L43

I think the `@game` local is only around for the duration of the request, but if someone accidentally did `@game.save!` that would update all "Game Lab" games  to "Sprite Lab".

Reply:
Sample commit for adding a new game type here: https://github.com/code-dot-org/code-dot-org/commit/13dccca4972a72d4daec6fb819e1444f258c70a7

It's kind of gross that we have to add it so many different places in that file. We may want to track as tech debt. Also that the model for Sprite Lab levels is called `gamelab_jr.rb`...

Reply:
What other implications/risks will splitting Spritelab out into its own game type have? 

I could also just override `game_display_name`, which I think is only used to set the tab name. This would eliminate the risk of modifying the @game record.

Reply:
Ah good question. You may also need to update this line to account for the new game type:

https://github.com/code-dot-org/code-dot-org/blob/f5af68767efcbaa1280ac3a71c4c16d7511d7be5/dashboard/app/helpers/levels_helper.rb#L297

It should probably be:
```rb
use_gamelab = @level.is_a?(Gamelab)
```
since `GamelabJr` inherits from `Gamelab`.

---

This line can also probably be deleted, now that we have a specific game type for Sprite Lab:

https://github.com/code-dot-org/code-dot-org/blob/f5af68767efcbaa1280ac3a71c4c16d7511d7be5/dashboard/app/helpers/levels_helper.rb#L776